src/validators/dataclass.rs

```diff
@@ -176,6 +178,28 @@ impl Validator for DataclassArgsValidator {
                     ($args:ident, $get_method:ident, $get_macro:ident, $slice_macro:ident) => {{
                         // go through fields getting the value from args or kwargs and validating it
                         for (index, field) in self.fields.iter().enumerate() {
+                            if (!field.init) {
+                                match field.validator.default_value(py, Some(field.name.as_str()), state) {
+                                    Ok(Some(value)) => {
+                                        // Default value exists, and passed validation if required
+                                        set_item!(field, value);
+                                    },
+                                    Ok(None) => continue,
+                                    Err(ValError::Omit) => continue,
+                                    Err(ValError::LineErrors(line_errors)) => {
+                                        for err in line_errors {
+                                            // Note: this will always use the field name even if there is an alias
+                                            // However, we don't mind so much because this error can only happen if the
+                                            // default value fails validation, which is arguably a developer error.
+                                            // We could try to "fix" this in the future if desired.
+                                            errors.push(err);
+                                        }
+                                    }
```

Comment:
```suggestion
                                    // Note: this will always use the field name even if there is an alias
                                    // However, we don't mind so much because this error can only happen if the
                                    // default value fails validation, which is arguably a developer error.
                                    // We could try to "fix" this in the future if desired.
                                    Err(ValError::LineErrors(line_errors)) => errors.extend(line_errors),
```